clients/client-java/src/main/java/com/datastrato/gravitino/client/MessagingCatalog.java

```diff
@@ -0,0 +1,226 @@
+            ErrorHandlers.topicErrorHandler());
+    resp.validate();
+
+    return resp.getTopic();
+  }
+
+  /**
+   * Create a new topic with the given identifier, comment, data layout and properties.
+   *
+   * @param ident A topic identifier.
+   * @param comment The comment of the topic object. Null is set if no comment is specified.
+   * @param dataLayout The message schema of the topic object. Always null because it's not
+   *     supported yet.
+   * @param properties The properties of the topic object. Empty map is set if no properties are
+   *     specified.
+   * @return The created topic object.
+   * @throws NoSuchSchemaException if the schema with specified namespace does not exist.
+   * @throws TopicAlreadyExistsException if the topic with specified identifier already exists.
+   */
+  @Override
+  public Topic createTopic(
+      NameIdentifier ident, String comment, DataLayout dataLayout, Map<String, String> properties)
+      throws NoSuchSchemaException, TopicAlreadyExistsException {
+    NameIdentifier.checkTopic(ident);
+
+    TopicCreateRequest req =
+        TopicCreateRequest.builder()
+            .name(ident.name())
+            .comment(comment)
+            .properties(properties)
+            .build();
+
+    TopicResponse resp =
+        restClient.post(
+            formatTopicRequestPath(ident.namespace()),
+            req,
+            TopicResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.topicErrorHandler());
+    resp.validate();
+
+    return resp.getTopic();
+  }
+
+  /**
+   * Alter the topic with the given identifier.
+   *
+   * @param ident A topic identifier.
+   * @param changes The changes to apply to the topic.
+   * @return The altered topic object.
+   * @throws NoSuchTopicException if the topic with the specified identifier does not exist.
+   * @throws IllegalArgumentException if the changes are invalid.
+   */
+  @Override
+  public Topic alterTopic(NameIdentifier ident, TopicChange... changes)
+      throws NoSuchTopicException, IllegalArgumentException {
+    NameIdentifier.checkTopic(ident);
+
+    List<TopicUpdateRequest> updates =
+        Arrays.stream(changes)
+            .map(DTOConverters::toTopicUpdateRequest)
+            .collect(Collectors.toList());
+    TopicUpdatesRequest updatesRequest = new TopicUpdatesRequest(updates);
+    updatesRequest.validate();
+
+    TopicResponse resp =
+        restClient.put(
+            formatTopicRequestPath(ident.namespace()) + "/" + ident.name(),
+            updatesRequest,
+            TopicResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.topicErrorHandler());
+    resp.validate();
+
+    return resp.getTopic();
+  }
+
+  /**
+   * Drop the topic with the given identifier.
+   *
+   * @param ident A topic identifier.
+   * @return True if the topic is dropped successfully, false the topic does not exist.
+   */
+  @Override
+  public boolean dropTopic(NameIdentifier ident) {
```

Comment:
@mchades @jerryshao In other classes, we use `try catch` to wrap drop method like GravitionAdminClient, RelationalCatalog. Why do we choose not to wrap? Is it expected?